onnxruntime/core/providers/cpu/cpu_execution_provider.cc

```diff
@@ -374,8 +374,10 @@
                                                       QuantizeLinear);
 class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 10, 12, int8_t,
                                                       QuantizeLinear);
-class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 10, uint8_t, QLinearMatMul);
-class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 10, int8_t, QLinearMatMul);
+class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 10, 20, uint8_t,
```

Comment:
>ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME [](http://example.com/codeflow?start=6&length=47)

`ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 10, uint8_t, QLinearMatMul);` enables QLinearMatMul for opset 10+. The new code is essentially equivalent to the old code. May I know what's the original problem?

Reply:
There's QLinearMatMul opset 21 from Onnx 1.16 which is not covered by Ort, we need to do it this way to cover opset 21.

Reply:
I see. QLinearMatMul changed schema in opset21.